Framework/API/src/FileFinder.cpp

```diff
@@ -207,8 +205,8 @@ FileFinderImpl::toInstrumentAndNumber(const std::string &hint) const {
     runPart = hint;
   } else {
     /// Find the last non-digit as the instrument name can contain numbers
-    std::string::const_reverse_iterator it = std::find_if(
-        hint.rbegin(), hint.rend(), std::not1(std::ptr_fun(isdigit)));
+    std::string::const_reverse_iterator it =
+        std::find_if(hint.rbegin(), hint.rend(), std::not_fn(isdigit));
```

Comment:
Could we take this opportunity to replace all of the `std::not_fn` with lambdas. That way they don't have to lookup `std::not_fn` if they aren't familiar:

`std::find_if(hint.rbegin(), hint.rend(), [](const char c){return !isdigit(c);});`

Reply:
Do you not think `not_fn(isdigit)` is more readable? I have to read further with the lambda and make sure I don't miss the `!` in there.

Reply:
🤔 Good point - it's finding a balance between dev's infrequently seeing `std::not_fn` and the boilerplate of a lambda

I think I'll agree with you on this one, as once you've looked up the various memory references the STL gives you it's obvious to read